packages/adapter-next/src/index.ts

```diff
@@ -21,7 +21,7 @@ export default defineSliceMachinePlugin<PluginOptions>({
 	defaultOptions: {
 		format: true,
 	},
-	setup({ hook }) {
+	setup({ actions: { hook } }) {
```

Comment:
`hook()` and `unhook()` are unique to the `setup` function, so I wonder if it should not be nested under `actions` and instead be adjacent to `actions`. Being able to call `hook` within another hook, for example, shouldn't be allowed (I think). (Of course, we can't stop someone from passing the reference, but it should be discouraged.)

By nesting it under `actions`, we are implying that `hook()` is available in the `actions` object everywhere else, which it currently is not.

This leads to the following questions:
- Is there a case where a hook needs to add an additional hook via `actions.hook()`?
- Is there a case where a hook needs to remove a hook via `actions.unhook()`?

Reply:
> This leads to the following questions:
> 
> * Is there a case where a hook needs to add an additional hook via `actions.hook()`?
> * Is there a case where a hook needs to remove a hook via `actions.unhook()`?

I don't think there's a usecase for that yet, I'll remove hooks from here and have them available top level